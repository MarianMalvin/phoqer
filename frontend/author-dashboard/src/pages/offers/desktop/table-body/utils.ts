import { TableCellSizeEnum } from 'phoqer';

export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'status', title: 'Status' },
    { key: 'options', title: 'Options' },
    { key: 'images', title: 'Images' },
    { key: 'title', title: 'Title' },
    { key: 'category', title: 'Category' },
    { key: 'reviews', title: 'Reviews' },
    { key: 'price', title: 'Price' },
    { key: 'sale_percentage', title: 'Sale percentage' },
    { key: 'sale_description', title: 'Sale description' },
    { key: 'description', title: 'Description' },
];

export const sizeMap: { [key: string]: TableCellSizeEnum } = {
    status: TableCellSizeEnum.Small,
    options: TableCellSizeEnum.Middle,
    images: TableCellSizeEnum.Small2Extra,
    title: TableCellSizeEnum.Large5Extra,
    description: TableCellSizeEnum.Large8Extra,
    sale_percentage: TableCellSizeEnum.MiddleExtra,
    sale_description: TableCellSizeEnum.Large4Extra,
    reviews: TableCellSizeEnum.Middle,
    price: TableCellSizeEnum.Middle,
    category: TableCellSizeEnum.Middle,
};
