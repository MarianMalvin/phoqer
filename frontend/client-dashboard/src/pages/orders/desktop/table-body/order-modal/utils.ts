export const cellMap = [
    { key: 'title', title: 'Title' },
    { key: 'status', title: 'Status' },
    { key: 'date', title: 'Created' },
    { key: 'expired', title: 'Expired' },
    { key: 'price', title: 'Price' },
    { key: 'category', title: 'Category' },
    { key: 'country', title: 'Country' },
    { key: 'city', title: 'City' },
    { key: 'zip', title: 'ZIP' },
    { key: 'address', title: 'Address' },
    { key: 'comment', title: 'Comment' },
    { key: 'sale', title: 'Sale' },
    { key: 'sale_description', title: 'Sale Description' },
];
