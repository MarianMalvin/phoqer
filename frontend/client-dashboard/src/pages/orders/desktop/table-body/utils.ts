import { TableCellSizeEnum } from 'phoqer';

export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'status', title: 'Status' },
    { key: 'options', title: 'Options' },
    { key: 'images', title: 'Images' },
    { key: 'title', title: 'Title' },
    { key: 'price', title: 'Price' },
    { key: 'user', title: 'Author' },
    { key: 'address', title: 'Address' },
    { key: 'comment', title: 'Comment' },
    { key: 'date', title: 'Created' },
    { key: 'expired', title: 'Expiration Date' },
];

export const sizeMap: { [key: string]: TableCellSizeEnum } = {
    status: TableCellSizeEnum.Small,
    options: TableCellSizeEnum.Middle,
    title: TableCellSizeEnum.Large3Extra,
    images: TableCellSizeEnum.Small2Extra,
    price: TableCellSizeEnum.Middle,
    address: TableCellSizeEnum.Large4Extra,
    comment: TableCellSizeEnum.Large3Extra,
    user: TableCellSizeEnum.LargeExtra,
    date: TableCellSizeEnum.Large2Extra,
    expired: TableCellSizeEnum.Large2Extra,
};
