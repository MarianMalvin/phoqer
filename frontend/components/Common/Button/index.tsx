import clsx from 'clsx';
import React, { MouseEvent, ReactElement } from 'react';
import { createUseStyles } from 'react-jss';

import Spinner from '../Preloaders/Spinner';

const useStyles = createUseStyles({
    loading: {
        pointerEvents: 'none',

        '& span': {
            opacity: 0,
        },
    },
    btn: {
        position: 'relative',
        fontSize: 'inherit',

        '& span': {
            fontSize: 'inherit',
        },
    },
    spinner: {
        position: 'absolute',
        top: '50%',
        left: '50%',
        transform: 'translate(-50%, -50%)',
        zIndex: 2,
    },
});

interface IProps {
    loading?: boolean;
    type?: 'button' | 'submit';
    className?: string;
    onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
    children: ReactElement | ReactElement[] | string | number;
}

const Button = ({ className, children, loading = false, onClick, type = 'button' }: IProps): ReactElement => {
    const css = useStyles();
    return (
        <button className={clsx(css.btn, className, loading && css.loading)} type={type} onClick={onClick}>
            {loading ? <Spinner className={css.spinner} /> : null}
            <span>{children}</span>
        </button>
    );
};

export default Button;
