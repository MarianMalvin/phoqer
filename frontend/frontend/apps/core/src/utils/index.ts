export * from './recentOffers';
export * from './errorHandler';
