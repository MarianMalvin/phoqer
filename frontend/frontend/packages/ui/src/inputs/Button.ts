import { ComponentStyleConfig } from '@chakra-ui/theme';

export const Button: ComponentStyleConfig = {
    baseStyle: {
        fontWeight: 500,
    },
    variants: {
        primary: {
            bg: 'common.black',
            border: '1px solid',
            borderColor: 'gray.400',
            color: 'common.white',
            borderRadius: '6px',
            svg: {
                fontSize: '1.4em',
            },

            _hover: {
                bg: 'gray.600',
            },

            _active: {
                borderColor: 'common.black',
                bg: 'common.black',
            },

            _dark: {
                bg: 'gray.100',
                borderColor: 'gray.100',
                color: 'common.black',

                _hover: {
                    bg: 'gray.400',
                },
                _active: {
                    bg: 'gray.300',
                },
            },
        },
        green: {
            bg: 'green.600',
            border: '1px solid',
            borderColor: 'green.600',
            color: 'common.white',
            svg: {
                fontSize: '1.4em',
            },

            _hover: {
                bg: 'green.500',
            },

            _active: {
                borderColor: 'common.black',
                bg: 'green.700',
            },
        },
        pill: {
            bg: 'primary.main',
            border: '1px solid',
            borderColor: 'primary.main',
            color: 'common.white',
            borderRadius: '3rem',

            _disabled: {
                bg: 'primary.main !important',

                _hover: {
                    bg: 'primary.main !important',
                },
            },

            svg: {
                fontSize: '1.4em',
            },

            _hover: {
                bg: 'primary.hover',
            },

            _active: {
                bg: 'common.black',
            },
        },
        link: {
            p: 0,
            bg: 'transparent',
            border: 'none',
            color: 'text.main',
            fontSize: '0.9em',
            fontWeight: 600,
            _hover: {
                color: 'gray.800',
            },
            _dark: {
                color: 'gray.500',

                _hover: {
                    color: 'gray.400',
                },
            },
        },
        secondary: {
            bg: 'common.white',
            border: '1px solid',
            borderColor: 'gray.500',
            color: 'common.black',
            borderRadius: '6px',

            _hover: {
                borderColor: 'gray.700',
            },
            _active: {
                borderColor: 'common.black',
            },

            _dark: {
                bg: 'dark.body',
                borderColor: 'gray.700',
                color: 'common.white',

                _hover: {
                    bg: 'gray.700',
                    borderColor: 'gray.100',
                },
                _active: {
                    borderColor: 'gray.400',
                },
            },
        },
        ghost: {
            position: 'relative',
            color: 'common.black',
            bg: 'transparent',

            _before: {
                content: '""',
                position: 'absolute',
                top: '50%',
                left: '50%',
                transform: 'translate(-50%, -50%)',
                height: '100%',
                width: '100%',
                borderRadius: 'md',
                bg: 'gray.100',
                zIndex: -1,
            },

            _after: {
                content: '""',
                position: 'absolute',
                top: '50%',
                left: '50%',
                transform: 'translate(-50%, -50%) scale(0)',
                height: '100%',
                width: '100%',
                borderRadius: 'md',
                bg: 'gray.300',
                zIndex: -1,
                transition: '0.1s ease-in-out',
            },

            _hover: {
                bg: 'transparent',
                _after: {
                    transform: 'translate(-50%, -50%) scale(1)',
                },
            },

            _active: {
                bg: 'transparent',
                _after: {
                    bg: 'gray.400',
                },
            },

            _dark: {
                color: 'common.white',

                _before: {
                    bg: 'gray.700',
                },

                _after: {
                    bg: 'gray.700',
                },

                _hover: {
                    _after: {
                        bg: 'gray.600',
                    },
                },

                _active: {
                    _after: {
                        bg: 'gray.500',
                    },
                },
            },
        },
        icon: {
            position: 'relative',
            borderRadius: '50%',
            color: 'common.black',

            svg: {
                position: 'relative',
                zIndex: 1,
                boxSize: '1.75em',
            },

            _before: {
                content: '""',
                position: 'absolute',
                top: '50%',
                left: '50%',
                transform: 'translate(-50%, -50%) scale(0)',
                height: '120%',
                width: '120%',
                borderRadius: '50%',
                bg: 'gray.200',
                transition: '0.1s ease-in-out',
            },

            _hover: {
                _before: {
                    transform: 'translate(-50%, -50%) scale(1)',
                },
            },

            _active: {
                _before: {
                    bg: 'gray.400',
                },
            },

            _dark: {
                color: 'common.white',

                _before: {
                    bg: 'gray.700',
                },

                _hover: {
                    _before: {
                        bg: 'gray.600',
                    },
                },

                _active: {
                    _before: {
                        bg: 'gray.500',
                    },
                },
            },
        },
        'icon-ghost': {
            position: 'relative',
            borderRadius: '50%',
            color: 'common.black',

            svg: {
                position: 'relative',
                zIndex: 1,
                boxSize: '1.75em',
            },

            _before: {
                content: '""',
                position: 'absolute',
                top: '50%',
                left: '50%',
                transform: 'translate(-50%, -50%) scale(0.8)',
                height: '120%',
                width: '120%',
                borderRadius: '50%',
                bg: 'gray.200',
                transition: '0.1s ease-in-out',
            },

            _hover: {
                _before: {
                    bg: 'gray.300',
                    transform: 'translate(-50%, -50%) scale(1)',
                },
            },

            _active: {
                _before: {
                    bg: 'gray.400',
                },
            },

            _dark: {
                color: 'common.white',

                _before: {
                    bg: 'gray.700',
                },

                _hover: {
                    _before: {
                        bg: 'gray.600',
                    },
                },

                _active: {
                    _before: {
                        bg: 'gray.500',
                    },
                },
            },
        },
    },
    sizes: {
        md: {
            fontSize: '0.85rem',
        },
        lg: {
            fontSize: '1rem',
        },
    },
    defaultProps: {
        size: 'sm',
        variant: 'primary',
    },
};
