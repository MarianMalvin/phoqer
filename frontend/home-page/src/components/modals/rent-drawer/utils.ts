import { RentPayload } from './type';

export const RENT_DRAWER_DATA_KEY = 'rent-drawer-data';

export const initialValues: RentPayload = {
    country: '',
    city: '',
    zip: '',
    address: '',
    comment: '',
};

export const getInitialValues = (): RentPayload => {
    try {
        const json = localStorage.getItem(RENT_DRAWER_DATA_KEY);
        if (!json) {
            return initialValues;
        }

        const data = JSON.parse(json);
        return data || initialValues;
    } catch {
        return initialValues;
    }
};

export const setInitialValues = (value: RentPayload): void => {
    try {
        localStorage.setItem(RENT_DRAWER_DATA_KEY, JSON.stringify(value));
    } catch (error) {
        console.log(error);
    }
};
