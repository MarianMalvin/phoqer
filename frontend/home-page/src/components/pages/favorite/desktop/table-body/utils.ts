export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'image', title: 'Image' },
    { key: 'options', title: 'Rent now' },
    { key: 'title', title: 'Title' },
    { key: 'price', title: 'Price' },
    { key: 'sale', title: 'Sale' },
    { key: 'category', title: 'Category' },
];

export const sizeMap: { [key: string]: number } = {
    image: 10,
    title: 45,
    price: 15,
    sale: 10,
    category: 12,
    options: 18,
};
