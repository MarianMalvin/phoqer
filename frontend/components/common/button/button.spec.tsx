import { fireEvent, screen } from '@testing-library/react';

import { render } from '../../../tests/tests.utils';
import Button from './index';

describe('Test Button component', () => {
    it('Should match snapshot', () => {
        const { asFragment } = render(<Button>test</Button>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Should match snapshot', () => {
        const { asFragment } = render(
            <Button className="test" type="submit" loading>
                test
            </Button>,
            {},
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('Test onClick', () => {
        const handleClick = jest.fn();
        render(<Button onClick={handleClick}>test</Button>, {});
        fireEvent.click(screen.getByRole(/button/i));
        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});
