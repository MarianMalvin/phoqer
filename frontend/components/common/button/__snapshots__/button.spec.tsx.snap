// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component Should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn-0-2-3"
    type="button"
  >
    <span>
      test
    </span>
  </button>
</DocumentFragment>
`;

exports[`Test Button component Should match snapshot 2`] = `
<DocumentFragment>
  <button
    class="btn-0-2-3 test loading-0-2-2"
    type="submit"
  >
    <div
      class="wrap-0-2-5 spinner-0-2-4"
    >
      <img
        alt="spinner"
        class="img-0-2-6"
        src="/spinner.gif"
      />
    </div>
    <span>
      test
    </span>
  </button>
</DocumentFragment>
`;
